### DISCUSSION TITLE
Non-strict privileged level

### DISCUSSION THREAD

**Discussion by denisbondar** [carlmontanari/scrapli#303] [Q&A]
There are some devices on my network that have privileged mode enabled by default, and after logging on to these devices there is no need to enter the enable command because we are already in this mode.
There are other devices where we get the unprivileged mode by default after logging on and we need to enter the `enable` command to switch to this mode.

There are commands that are available in either of these modes and can be executed in both unprivileged and privileged modes in the same way.

However, it is not entirely clear how to configure the driver. If you set `privilege_exec` to `default_desired_privilege_level`, it will always try to go to privileged mode, and if you set `exec`, it will always try to go back to unprivileged mode.

My question is:
**Is there any way to use the default unprivileged mode so that if the default device provides privileged mode, Scraply will not try to switch to unprivileged mode by passing disable or other command specified in deescalate?**

**Comment by carlmontanari** [comment]
I think you can just use the “generic driver” with a custom “on open” function to do whatever setup steps you have. If you login to the device and then decide it should be in “privileged” mode (meaning you actually want the “network driver”), you can use the “commandeer” function to basically change from the network drive to generic (you could do this in reverse too of course). Hopefully that makes sense! Am on mobile at the moment but I believe there is a commandeer example and for the custom in open it’s just a callable that accepts the cls/self if the driver as first and only arg— so you can do whatever ya need to with that.

**Reply by denisbondar** [reply]
Thank you. This issue has now been put on the back burner because of other, more important issues. I will definitely come back to it a little later to try to do what you recommended.
